src/main/java/core/basesyntax/WorkWithFile.java

```
@@ -1,7 +1,64 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
-    public void getStatistic(String fromFileName, String toFileName) {
+    private static final String SUPPLY_LABEL = "supply";
+    private static final String BUY_LABEL = "buy";
+    private static final String RESULT_LABEL = "result";
+    private static final String DELIMITER = ",";
+    private static final String NEW_LINE = "\n";
 
+    private int[] readAndAggregate(String fromFileName) {
+        int[] result = new int[2];
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            String line = reader.readLine();
+            String[] key = new String[2];
+            while (line != null) {
+                if (key.length == 2) {
+                    key = line.split(DELIMITER);
+                    if (key[0].equals(SUPPLY_LABEL)) {
+                        result[0] += Integer.parseInt(key[1]);
+                        line = reader.readLine();
+                    } else if (key[0].equals(BUY_LABEL)) {
+                        result[1] += Integer.parseInt(key[1]);
```

You call `Integer.parseInt(key[1])` without trimming the token and without catching NumberFormatException to wrap it with a descriptive RuntimeException including the file name and offending line. Tests with whitespace or invalid numbers will produce unclear exceptions.

This violates checklist item: "Don't ignore exceptions". Suggested change: `String amountToken = tokens[1].trim(); try { int amount = Integer.parseInt(amountToken); } catch (NumberFormatException e) { throw new RuntimeException("Can't parse amount '" + amountToken + "' in file " + fromFileName + " for line '" + line + "'", e); }`